src/components/ApplicationGroup/AppGroupDetailsRoute.tsx

```diff
@@ -15,29 +15,68 @@ import { EMPTY_LIST_MESSAGING, ENV_APP_GROUP_GA_EVENTS, NO_ACCESS_TOAST_MESSAGE
 import { ReactComponent as Settings } from '../../assets/icons/ic-settings.svg'
 import { getEnvAppList } from './AppGroup.service'
 import { AppGroupAdminType, EnvHeaderType } from './AppGroup.types'
+import { getAppList } from '../app/service'
+import { MultiValue } from 'react-select'
+import { OptionType } from '../app/types'
+import AppGroupAppFilter from './AppGroupAppFilter'
 import EnvCIDetails from './Details/EnvCIDetails/EnvCIDetails'
 import EnvCDDetails from './Details/EnvCDDetails/EnvCDDetails'
 import '../app/details/app.css'
 
+const AppGroupAppFilterContext = React.createContext(null)
+
+export function useAppGroupAppFilterContext() {
+    const context = React.useContext(AppGroupAppFilterContext)
+    if (!context) {
+        throw new Error(`cannot be rendered outside the component`)
+    }
+    return context
+}
+
 export default function AppGroupDetailsRoute({ isSuperAdmin }: AppGroupAdminType) {
     const { path } = useRouteMatch()
     const { envId } = useParams<{ envId: string }>()
     const [envName, setEnvName] = useState('')
-    const [showEmpty, setShowEmpty] = useState<boolean>(true)
+    const [showEmpty, setShowEmpty] = useState<boolean>(false)
+    const [appListLoading, setAppListLoading] = useState<boolean>(false)
     const [loading, envList] = useAsync(getEnvAppList, [])
+    const [appListOptions, setAppListOptions] = useState<OptionType[]>([])
+    const [selectedAppList, setSelectedAppList] = useState<MultiValue<OptionType>>([])
 
     useEffect(() => {
         if (envList?.result) {
             const environment = envList.result.envList?.find((env) => env.id === +envId)
             setEnvName(environment.environment_name)
             setShowEmpty(!environment.appCount)
         }
-    }, [envList])
+    }, [envList, envId])
 
-    const renderEmptyAndLoading = () => {
-        if (loading) {
-            return <Progressing pageLoader />
+    const getAppListData = async (): Promise<void> => {
+        setAppListLoading(true)
+        const { result } = await getAppList({ environments: [+envId] })
+        if (result.appContainers?.length) {
+            setAppListOptions(
+                result.appContainers
+                    .map((appDetails) => {
+                        return {
+                            value: appDetails.appId,
+                            label: appDetails.appName,
+                        }
+                    })
+                    .sort(sortOptionsByLabel),
+            )
         }
+        setAppListLoading(false)
+    }
+
+    useEffect(() => {
```

Comment:
useEffect must move above this component

Reply:
Done